src/common/pf_snmp.c

```diff
@@ -134,22 +134,80 @@ static void pf_snmp_encode_signal_delays (
 
 void pf_snmp_get_system_name (pnet_t * net, pf_snmp_system_name_t * name)
 {
+   const char * directory = net->fspm_cfg.file_directory;
    int error;
 
    CC_STATIC_ASSERT (sizeof (name->string) >= PNAL_HOSTNAME_MAX_SIZE);
 
-   error = pnal_get_hostname (name->string);
+   error = pf_file_load (directory, PF_FILENAME_SYSNAME, name, sizeof (*name));
    if (error)
    {
-      name->string[0] = '\0';
+      error = pnal_get_hostname (name->string);
+      if (error)
+      {
+         name->string[0] = '\0';
+      }
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): Could not yet read sysName from nvm.\n",
```

Comment:
Maybe
`"SNMP(%d): Could not yet read sysName from nvm. Using hostname \"%s\".\n"`

Reply:
Done.